odoo/addons/test_lint/tests/test_checkers.py

```diff
@@ -458,3 +458,76 @@ def wrapper2(var):
         """)
         with self.assertMessages("sql-injection"):
             checker.visit_call(list(node.get_children())[1])
+
+
+@unittest.skipUnless(pylint and pylint_bin, "testing lints requires pylint")
+class TestI18nChecks(TransactionCase):
+
+    def check(self, test_content):
+        with tempfile.NamedTemporaryFile(mode="w", encoding="utf-8", delete=False) as f:
+            self.addCleanup(os.remove, f.name)
+            f.write(dedent(test_content).strip())
+        res = run(
+            [
+                pylint_bin,
+                f"--rcfile={os.devnull}",
+                "--load-plugins=_odoo_checker_gettext",
+                "--disable=all",
+                "--enable=gettext-variable,gettext-placeholders,gettext-repr",
+                "--output-format=json",
+                f.name,
+            ],
+            stdout=PIPE,
+            encoding="utf-8",
+            env={
+                **os.environ,
+                "PYTHONPATH": HERE + os.pathsep + os.environ.get("PYTHONPATH", ""),
+            },
+            check=False,
+        )
+        return res.returncode, json.loads(res.stdout)
+
+    def test_gettext_variable(self):
+        exit_code, errors = self.check(
+            """
+            some_variable = "Roblox Mini Golf! [ACTUALLY FIXED]"
+            _(some_variable)
+            _lt(513)
+            _lt("string but" + "not static")
+            """
+        )
+        self.assertNotEqual(exit_code, os.EX_OK)
+        self.assertEqual(len(errors), 3)
+        for error in errors:
+            self.assertEqual(error["symbol"], "gettext-variable")
+
+    def test_gettext_placeholders(self):
+        exit_code, errors = self.check(
+            """
+            _("shouldn't match escaped %%s %%s")
+            """
+        )
+        self.assertEqual(exit_code, os.EX_OK)
+        self.assertFalse(errors)
+        exit_code, errors = self.check(
+            """
+            _("more than one unnamed placeholder: %s %s")
```

Comment:
Just wondering, what if we have multiple placeholders, but only one is unnamed? 🤔

Reply:
Not caught. That said, you can't have both args and kwargs: it would crash to begin with